File: rb/features.rb
# Jsonplaceholder SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module JsonplaceholderFeatures
  def self.make_feature(name)
    case name
    when "base"
      JsonplaceholderBaseFeature.new
    when "test"
      JsonplaceholderTestFeature.new
    else
      JsonplaceholderBaseFeature.new
    end
  end
end

File: py/jsonplaceholder_sdk/features.py
# Jsonplaceholder SDK feature factory

from jsonplaceholder_sdk.feature.base_feature import JsonplaceholderBaseFeature
from jsonplaceholder_sdk.feature.test_feature import JsonplaceholderTestFeature


def _make_feature(name):
    features = {
        "base": lambda: JsonplaceholderBaseFeature(),
        "test": lambda: JsonplaceholderTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
